llvm/include/llvm/Transforms/Utils/Local.h

```diff
@@ -498,6 +498,7 @@ LLVM_ABI void dropDebugUsers(Instruction &I);
 ///
 /// The moved instructions receive the insertion point debug location values
 /// (DILocations) and their debug intrinsic instructions are removed.
+/// Selects and indirect calls keep their MD_prof metadata.
```

Comment:
I believe this is different from https://github.com/llvm/llvm-project/pull/152420#discussion_r2264594682 because this is less generic - there, it was about whether a caller asking to `dropUnknown...Metadata` would be surprised to get `MD_prof` still there. Here, I'd argue the API is specifically about hoisting, and it's more reasonable to keep around some metadata. @nikic

Reply:
Done, ptal.